server/controllers/submission.js

```diff
@@ -19,25 +19,132 @@ exports.getAllSubmissions = asyncHandler(async (req, res, next) => {
+          submissions.forEach((submission) => {
+            const submissionData = {
+              _id: submission._id,
+              name: submission.userId.username,
+              files: submission.files,
+              ownerName: contestData.userId.username,
+              title: contestData.title,
+              description: contestData.description,
+              prizeAmount: contestData.prizeAmount,
+              profilePicUrl: contestData.userId.profilePicUrl,
+            };
+            submissionList.push(submissionData);
+          });
+        } else {
+          const submissionData = {
+            _id: null,
+            name: null,
+            files: null,
+            ownerName: contestData.userId.username,
+            title: contestData.title,
+            description: contestData.description,
+            prizeAmount: contestData.prizeAmount,
+            profilePicUrl: contestData.userId.profilePicUrl,
+          };
+          submissionList.push(submissionData);
+        }
+
+        res.status(200).json({
+          submission: submissionList,
+          isOwner: true,
+        });
+      } else {
+        const submission = await Submission.findOne({ userId: userId }).populate('userId');
+
+        if (submission) {
+          if (submission.contestId.toString() === contestId) {
+            const submissionData = {
+              _id: submission._id,
+              name: submission.userId.username,
+              files: submission.files,
+              ownerName: contestData.userId.username,
+              title: contestData.title,
+              description: contestData.description,
+              prizeAmount: contestData.prizeAmount,
+              profilePicUrl: contestData.userId.profilePicUrl,
+            };
+            submissionList.push(submissionData);
+          } else {
+            const submissionData = {
+              _id: null,
+              name: null,
+              files: null,
+              ownerName: contestData.userId.username,
+              title: contestData.title,
+              description: contestData.description,
+              prizeAmount: contestData.prizeAmount,
+              profilePicUrl: contestData.userId.profilePicUrl,
+            };
+            submissionList.push(submissionData);
+          }
+        } else {
+          const submissionData = {
+            _id: null,
+            name: null,
+            files: null,
+            ownerName: contestData.userId.username,
+            title: contestData.title,
+            description: contestData.description,
+            prizeAmount: contestData.prizeAmount,
+            profilePicUrl: contestData.userId.profilePicUrl,
+          };
+          submissionList.push(submissionData);
+        }
+
+        res.status(200).json({ submission: submissionList, isOwner: false });
+      }
+    } else {
+      const submission = await Submission.findOne({ userId: userId }).populate('userId');
```

Comment:
Isn't this the same as the if/else clause beginning on Line 61?

Reply:
there are in difference if/else block. the reason is to check the situations: 1, the user is not the contest owner and has no submissions. 2, he is not the owner but he has submissions. 3, he is the owner but no submission. 4, he is the owner and he has submissions

Reply:
As we discussed on our call this morning, if a user is a contest creator - they should not have the ability to submit designs to their own contest (so you will need to change this accordingly)